### PR TITLE
Release/v 1.0.0

### PR DESCRIPTION
**Pull request by shuds13**
Ref  #1008

- [x] Version update
- [x] setup.py development status to production
- [x] update wci.yml
- [x] Release notes
- [x] Docs work
- [x] Testing done

### CODE REVIEW

File: .wci.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: libEnsemble
 icon: https://raw.githubusercontent.com/Libensemble/libensemble/main/docs/images/libE_logo.png
 headline: Tool for running dynamic ensembles.
 description: |
-  libEnsemble is a Python toolkit for coordinating workflows of asynchronous
-  and dynamic ensembles of calculations.
+  libEnsemble is a Python toolkit for running workflows dynamic ensembles
```
Comment (wildsm) [inline .wci.yml:5]: What is this supposed to read?

Comment (shuds13) [reply .wci.yml]: Fixed

**Comment by wildsm** [comment]
Given the change from funcx to Globus Compute, a number of changes may be in order. Examples:

- [ ] update and rename the image docs/images/funcxmodel.png
- [ ] update and rename the scaling test/path libensemble/tests/scaling_tests/forces/funcx_forces
- [ ] change the funcx hyperlink referenced in docs/introduction_latex.rst to globus compute (I would have done this but was annoyed by the sequence of redirects experienced and think we need a consistent reference point for globus compute given the changes.

I suggest reviewing `grep funcx -ir`

**Comment by wildsm** [comment]
I have done a pass over the diffs in the 183 files. 
Thanks all for the big push.

**Comment by shuds13** [comment]
> update and rename the image docs/images/funcxmodel.png

Image name updated. The image itself will be updated in future release.

File: docs/running_libE.rst [hunk 2]
```diff
@@ -291,10 +284,16 @@ For example::
     os.environ["OMP_NUM_THREADS"] = 4
 
 set in your simulation script before the Executor submit command will export the setting to your run.
```
Comment (jmlarson1) [inline docs/running_libE.rst:286]: Can we format the phrase "Executor submit" in some way?

File: libensemble/resources/scheduler.py [hunk 3]
```diff
@@ -46,13 +46,13 @@ def __init__(self, user_resources=None, sched_opts={}):
...
                 Try to split resource sets across more nodes if space is not currently
                 available on the minimum node count required. Allows more efficient
                 scheduling.
                 Default: True
 
-            "match_slots" [boolean]:
+            "match_slots" [Boolean]:
                 When splitting resource sets across multiple nodes, slot IDs must match.
                 Useful if setting an environment variable such as ``CUDA_VISIBLE_DEVICES``
                 to specific slots counts, which should match over multiple nodes.
```
Comment (jmlarson1) [inline libensemble/resources/scheduler.py:58]: "slot counts" or "slots counts"?

File: libensemble/tests/functionality_tests/test_1d_sampling_with_profile.py [hunk 4]
```diff
@@ -67,11 +69,11 @@
...
             assert file in os.listdir(), "Expected profile {file} not found after run"
-            with open(file, "r") as f:
-                data = f.read().split()
-                num_worker_funcs_profiled = sum(["worker" in i for i in data])
-            assert num_worker_funcs_profiled >= 8, (
-                "Insufficient number of " + "worker functions profiled: " + str(num_worker_funcs_profiled)
-            )
-
-            os.remove(file)
+            # with open(file, "r") as f:
```
Comment (jmlarson1) [inline libensemble/tests/functionality_tests/test_1d_sampling_with_profile.py:72]: Did we want this commented out on purpose?

File: libensemble/tests/regression_tests/test_1d_sampling.py [hunk 5]
```diff
@@ -21,28 +21,28 @@
...
-    sampling.add_random_streams()
+    sampling.persis_info = add_unique_random_streams({}, sampling.nworkers + 1)
     sampling.exit_criteria = ExitCriteria(gen_max=501)
 
     sampling.run()
     if sampling.is_manager:
         assert len(sampling.H) >= 501
         print("\nlibEnsemble with random sampling has generated enough points")
-        sampling.save_output(__file__)
+    sampling.save_output(__file__)
```
Comment (jmlarson1) [inline libensemble/tests/regression_tests/test_1d_sampling.py:48]: Shouldn't only the manager save (H is usually empty for workers)

Comment (jlnav) [reply libensemble/tests/regression_tests/test_1d_sampling.py:48]: Yes, only the manager saves, but `Ensemble.save_output()` doesn't need to be:

```python

    def save_output(self, file: str):
        """
        Writes out History array and persis_info to files.
        If using a workflow_dir, will place with specified filename in that directory.

        Format: ``<calling_script>_results_History_length=<length>_evals=<Completed evals>_ranks=<nworkers>``
        """
        if self.is_manager:
            if self._get_option("libE_specs", "workflow_dir_path"):
                save_libE_output(
                    self.H, self.persis_info, file, self.nworkers, dest_path=self.libE_specs.workflow_dir_path
                )
            else:
                save_libE_output(self.H, self.persis_info, file, self.nworkers)
```

**Review by jlnav — APPROVED** [review]

File: libensemble/tests/unit_tests_nompi/test_aaa_comms.py [hunk 6]
```diff
@@ -52,42 +53,42 @@ def test_qcomm_proc_terminate2():
...
-        try:
-            mgr_comm.terminate(timeout=0.5)
-            flag = False
-        except comms.Timeout:
-            pass
-        assert flag, "Should time out on terminate"
-
-        assert mgr_comm.running, "Should still be running"
-        mgr_comm.send("Done")
+# def ignore_handler(signum, frame):
```
Comment (jmlarson1) [inline libensemble/tests/unit_tests_nompi/test_aaa_comms.py:56]: Intentionally commented out?

**Review by jmlarson1 — APPROVED** [review]

File: libensemble/tests/unit_tests_nompi/test_aaa_comms.py (hunk 6 above)
Comment (shuds13) [reply libensemble/tests/unit_tests_nompi/test_aaa_comms.py:56]: This is the test we removed that is problematic on CI (macOS in virtual env). We could remove it.

Comment (jmlarson1) [reply libensemble/tests/unit_tests_nompi/test_aaa_comms.py:56]: We could remove, or at least add a note to the source code about why it's commented out

File: docs/running_libE.rst (hunk 2 above)
Comment (shuds13) [reply docs/running_libE.rst]: done

File: libensemble/resources/scheduler.py (hunk 3 above)
Comment (shuds13) [reply libensemble/resources/scheduler.py]: just 'slots'. I changed that.

**Pull request merged**